ckanext/validation/helpers.py

```diff
@@ -67,6 +72,18 @@ def get_validation_badge(resource, in_listing=False):
         title=resource.get('validation_timestamp', ''))
 
 
+def _get_schema_or_default_schema(resource):
+    schema = resource.get('schema')
```

Comment:
This line is a waste of time if line 79 is executed. Perhaps it should be in an `else` block.